crates/host/src/wasmbus/mod.rs

```
@@ -978,14 +985,79 @@ impl Host {
             ctl_nats_url = config.ctl_nats_url.as_str(),
             "connecting to NATS control server"
         );
-        let nats = async_nats::connect(config.ctl_nats_url.as_str())
-            .await
-            .context("failed to connect to NATS control server")?;
+        let ctl_nats = match (config.ctl_jwt.as_ref(), config.ctl_seed.as_ref()) {
+            (Some(jwt), Some(seed)) => {
```

We should probably return an error if only one of these is set